types/roslib/index.d.ts

```diff
@@ -575,6 +578,40 @@ declare namespace ROSLIB {
         normalize(): void;
     }
 
+    export class Pose {
+        position: Vector3;
+        orientation: Quaternion;
+
+        /**
+         * A Pose in 3D space. Values are copied into this object.
+         *
+         *  @constructor
+         *  @param options - object with following keys:
+         *   * position - the Vector3 describing the position
+         *   * orientation - the ROSLIB.Quaternion describing the orientation
+         */
+        constructor(
+            options?: {
+                position?: ConstructorParameters<typeof Vector3>[0] | null;
+                orientation?: ConstructorParameters<typeof Quaternion>[0] | null;
+            } | null,
+        );
```

Comment:
I also have an alternative option that declares
```ts
interface Vector3Like {
    x?: number | null;
    y?: number | null;
    z?: number | null;
}
interface QuaternionLike {
    x?: number | null;
    y?: number | null;
    z?: number | null;
    w?: number | null;
}
```
And then uses `Vector3Like` instead of `ConstructorParameters<typeof Vector3>[0]` if that is preferred.

Reply:
That seems more accurate yes, as Pose seems to be always converting to proper Vector3/Quaternion. So yes, I would suggest going with that change.